packages/swapper/src/swappers/ChainflipSwapper/endpoints.ts

```diff
@@ -1,28 +1,193 @@
+      to: depositAddress,
+      value: step.sellAmountIncludingProtocolFeesCryptoBaseUnit,
+      chainSpecific: {
+        from,
+        contractAddress: isTokenSend ? assetReference : undefined,
+        data: undefined,
+      },
+      sendMax: false,
+    }
+    const feeData = await adapter.getFeeData(getFeeDataInput)
+    const fees = feeData[FeeDataKey.Average]
+
+    if (!isExecutableTradeStep(step)) throw Error('Unable to execute trade step')
+
+    const unsignedTxInput = await adapter.buildSendApiTransaction({
+      to: depositAddress,
+      from,
+      value: step.sellAmountIncludingProtocolFeesCryptoBaseUnit,
+      accountNumber: step.accountNumber,
+      chainSpecific: {
+        gasLimit: fees.chainSpecific.gasLimit,
+        contractAddress: isTokenSend ? assetReference : undefined,
+        ...(supportsEIP1559
+          ? {
+              maxFeePerGas: fees.chainSpecific.maxFeePerGas!,
+              maxPriorityFeePerGas: fees.chainSpecific.maxPriorityFeePerGas!,
+            }
+          : {
+              gasPrice: fees.chainSpecific.gasPrice,
+            }),
+      },
+    })
+
+    return {
+      chainId: Number(fromChainId(chainId).chainReference),
+      data: unsignedTxInput.data,
+      to: unsignedTxInput.to,
+      from,
+      value: unsignedTxInput.value,
+      gasLimit: unsignedTxInput.gasLimit,
+      maxFeePerGas: unsignedTxInput.maxFeePerGas,
+      maxPriorityFeePerGas: unsignedTxInput.maxPriorityFeePerGas,
+      gasPrice: unsignedTxInput.gasPrice,
+    }
+  },
   getUnsignedUtxoTransaction,
-  checkTradeStatus: (
-    _input,
-  ): Promise<{
+  checkTradeStatus: async ({
+    config,
+    quoteId,
+  }): Promise<{
     status: TxStatus
     buyTxHash: string | undefined
     message: string | [string, InterpolationOptions] | undefined
   }> => {
-    // TODO(gomes): implement me
+    const swap = tradeQuoteMetadata.get(quoteId)
+    if (!swap) throw Error(`missing trade quote metadata for quoteId ${quoteId}`)
+    // Note, the swapId isn't the quoteId - we set the swapId at pre-execution time, when getting the receive addy and instantiating a flip swap
+    const swapId = swap.id
+
+    const brokerUrl = config.REACT_APP_CHAINFLIP_API_URL
+    const apiKey = config.REACT_APP_CHAINFLIP_API_KEY
+
+    const maybeStatusResponse = await chainflipService.get<ChainFlipStatus>(
+      `${brokerUrl}/status-by-id?apiKey=${apiKey}&swapId=${swapId}`,
+    )
+
+    if (maybeStatusResponse.isErr()) {
+      return {
+        buyTxHash: undefined,
+        status: TxStatus.Unknown,
+        message: undefined,
+      }
+    }
+
+    const { data: statusResponse } = maybeStatusResponse.unwrap()
+    const {
+      status: { state, swapEgress },
+    } = statusResponse
+
+    // Assume any state that isn't completed is pending.
+    // Going forward, we will want to add granular status polling similar to what we do for THOR.
+    if (state !== 'completed' || !swapEgress?.transactionReference) {
```

Comment:
`completed` or `failed` imho

Reply:
Now irrelevant since we use egress transactionReference as heuristics 🙏🏽